src/applications/MachineShopSimulator.java

```diff
@@ -47,38 +47,37 @@ boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
      * 
      * @return last job run on this machine
      */
-    Job changeState(int theMachine) {// Task on theMachine has finished,
+    private Job changeState(int theMachine) {// Task on theMachine has finished,
                                             // schedule next one.
         Job lastJob;
-        if (machine[theMachine].getActiveJob() == null) {// in idle or change-over
+        if (machine[theMachine].activeJob == null) {// in idle or change-over

```

Comment:
This (and the change below on line 57) are only possible because you made those fields visible up above. Like @deglm006, I'm not convinced that making them more visible is a great move, which then affects these changes.

Reply:
This is very fair and has been reverted with the commit that has changes based on Joe's suggestions.